sdk/core/azure-core-test/src/main/java/com/azure/core/test/InterceptorManager.java

```diff
@@ -96,13 +98,15 @@ public InterceptorManager(String testName, TestMode testMode) {
      * @throws NullPointerException If {@code testName} is {@code null}.
      */
     public InterceptorManager(TestContextManager testContextManager) {
-        this(testContextManager.getTestName(), testContextManager.getTestMode(), testContextManager.doNotRecordTest());
+        this(testContextManager.getTestName(), testContextManager.getTestPlaybackRecordingName(),
+            testContextManager.getTestMode(), testContextManager.doNotRecordTest());
     }
 
-    private InterceptorManager(String testName, TestMode testMode, boolean doNotRecord) {
+    private InterceptorManager(String testName, String playbackRecordName, TestMode testMode, boolean doNotRecord) {
         Objects.requireNonNull(testName, "'testName' cannot be null.");
 
         this.testName = testName;
+        this.playbackRecordName = playbackRecordName;
```

Comment:
If playbackRecordName is null, should this fallback to `testName`?

Reply:
Yes, everything calling into this should be passing `testName` for `playbackRecordName` though.